src/to_number.cc

```diff
@@ -0,0 +1,99 @@
+/**
+ * \file    to_number.cc
+ * \brief   Implementation of pow10().
+ * \authors \ref contributors
+ * \date    Created on 19 Februar 2024
+ *
+ * \copyright Copyright 2024 Deutsches Elektronen-Synchrotron (DESY), Hamburg
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 2.1 of the license, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <float.h>
+
+#include "gul14/to_number.h"
+
+namespace gul14 {
+
+namespace detail {
+
+long double pow10(int exponent)
```

Comment:
I think this can be made constexpr

Reply:
As `constexpr` implies `inline` this would not work for the exported function. It can only be exported _or_ inline (i.e. constexpr), not both.

At least I believe that.

We could make it inline of course, but that would instantiate it multiple times later on, and with the data tables I believe that is not a so good idea. I'm not really sure about the concrete downsides.
But the function will not be called in a constexpr context from within to_number, and so constexpr does not gain us anything. Could be different if we pull it out of the detail namespace.

Leaving this as is for the time being.
This is related to the question if we want this as template, below.